File: CSAdmin/Pantallas/TelPopUp.vb
Imports System.Data.Objects
Imports System.Data.Objects.DataClasses
Imports System.Data.EntityClient

Public Class TelPopUp
    Friend aux As Integer = 0
    Friend persona As New CSAdmin.Personas
    Private soloLectura As Boolean

    Private Sub TelPopUp_Load(sender As System.Object, e As System.EventArgs) Handles MyBase.Load
        Try
            If soloLectura Then
                MaskedTextBoxTel.Enabled = False
                ComboBoxTipo.Enabled = False
                ButtonAgregar.Enabled = False
            Else
                MaskedTextBoxTel.Enabled = True
                ComboBoxTipo.Enabled = True
                ButtonAgregar.Enabled = True
                ComboBoxTipo.SelectedIndex = 0
            End If
            Dim personasTelQuery As ObjectQuery(Of PersonasTel) = _
                   Main.BDContext.PersonasTel.Where("it.IdPersona = @Id")
            personasTelQuery.Parameters.Add(New ObjectParameter("Id", aux))
            DataGridViewTel.DataSource = personasTelQuery.Select("it.Tipo, it.Numero")
        Catch ex As Exception
            MessageBox.Show(ex.Message)
        End Try
    End Sub

    Private Sub TelPopUp_KeyPress(sender As System.Object, e As System.Windows.Forms.KeyPressEventArgs) Handles MyBase.KeyPress
        If e.KeyChar = ChrW(Keys.Escape) Then
            Me.Close()
        End If
    End Sub

    Private Sub ButtonAgregar_Click(sender As System.Object, e As System.EventArgs) Handles ButtonAgregar.Click
        Try
            Dim nuevoTel As New CSAdmin.PersonasTel
            With nuevoTel
                .Numero = MaskedTextBoxTel.Text
                .Tipo = ComboBoxTipo.SelectedItem
            End With
            persona.PersonasTel.Add(nuevoTel)
        Catch ex As Exception
            MessageBox.Show(ex.Message)
        End Try
    End Sub

    Public Sub New(ByRef per As CSAdmin.Personas, ByVal modo As Boolean)

        ' Llamada necesaria para el diseñador.
        InitializeComponent()

        ' Agregue cualquier inicialización después de la llamada a InitializeComponent().


        ' Se pasa por referencia el objeto para poder agregar los telefonos a la clase
        persona = per
        soloLectura = modo
    End Sub
End Class

File: CSAdminApp/Pantallas/telPopUp.cs
using CSAdminApp.Clases;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Data.Objects;
using System.Data.Objects.DataClasses;
using System.Data.EntityClient;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;

namespace CSAdminApp.Pantallas
{
    public partial class telPopUp : Form
    {
        private string[] tel;
        

        public telPopUp(string[] telefonos)
        {
            InitializeComponent();
            tel = telefonos;
        }

        private void telPopUp_Load(object sender, EventArgs e)
        {
            try
            {
                maskedTextBoxTel.Text = tel[0];
                maskedTextBoxCel.Text = tel[1];
            }
            catch (Exception ex)
            {
                MessageBox.Show(ex.Message);
            }
        }

        //private void buttonAgregar_Click(object sender, EventArgs e)
        //{
        //    try
        //    {
        //        // TODO Revisar restricciones telefonos
        //        PersonasTel nuevoTel = new PersonasTel();
        //        nuevoTel.Numero = Decimal.Parse(maskedTextBoxTel.Text);
        //        nuevoTel.Tipo = comboBoxTipo.SelectedItem.ToString();

        //        persona.PersonasTel.Add(nuevoTel);
        //        MessageBox.Show("Se agregó el numero de Tel: " + nuevoTel.Numero.ToString());
        //        FunmPC.limpiarForm(splitContainer.Panel1);
        //        maskedTextBoxTel.Focus();

        //        // TODO Refresco dinamico de datagrid
        //        bs.ResetBindings(false);
        //    }
        //    catch (Exception ex)
        //    {
        //        MessageBox.Show(ex.Message);
        //    }
        //}

        private void telPopUp_KeyPress(object sender, KeyPressEventArgs e)
        {
            if (e.KeyChar == Convert.ToChar(Keys.Escape))
            {
                this.Close();
            }
        }

        private void buttonOk_Click(object sender, EventArgs e)
        {
            try
            {
                tel[0] = maskedTextBoxTel.Text;
                tel[1] = maskedTextBoxCel.Text;
                this.Close();
            }
            catch (Exception ex)
            {
                MessageBox.Show(ex.Message);
            }
        }

        private void buttonCancelar_Click(object sender, EventArgs e)
        {
            this.Close();
        }
    }
}
